docs/configuration.md

```diff
@@ -0,0 +1,54 @@
+# Configuration
+
+A lot of code in YARP's repository is templated from a single configuration file, [config.yml](../config.yml). This file is used to generate the following files:
+
+* `ext/yarp/api_node.c` - for defining how to build Ruby objects for the nodes out of C structs
+* `include/yarp/ast.h` - for defining the C structs that represent the nodes
+* `java/org/yarp/AbstractNodeVisitor.java` - for defining the visitor interface for the nodes in Java
+* `java/org/yarp/Loader.java` - for defining how to deserialize the nodes in Java
+* `java/org/yarp/Nodes.java` - for defining the nodes in Java
+* `lib/yarp/node.rb` - for defining the nodes in Ruby
+* `lib/yarp/serialize.rb` - for defining how to deserialize the nodes in Ruby
+* `src/node.c` - for defining how to free the nodes in C and calculate the size in memory in C
+* `src/prettyprint.c` - for defining how to prettyprint the nodes in C
+* `src/serialize.c` - for defining how to serialize the nodes in C
+* `src/token_type.c` - for defining the names of the token types
+
+Whenever the structure of the nodes changes, you can run `rake templates` to regenerate these files. Alternatively tasks like `rake test` should pick up on these changes automatically. `config.yml` has a couple of top level fields, which we'll describe below.
+
```

Comment:
Should we add a line saying that all templated files will say they're templated at the top?

Reply:
👍